lib/connection/connection_config.js

```diff
@@ -260,17 +221,29 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
         protocol: proxyProtocol,
         noProxy: noProxy
       };
-    } else {
-      proxy =
-      {
-        host: proxyHost,
-        port: proxyPort,
-        protocol: proxyProtocol,
-        noProxy: noProxy
-      };
+    } 
+  
+    if (!proxy && GlobalConfig.isEnvProxyActive()) {
+      let proxyEnv = Util.getEnvVar('http_proxy') || Util.getEnvVar('https_proxy');
+
+      if (proxyEnv) {
+        proxyEnv = new URL(proxyEnv);
+        proxy = {
+          host: Util.validateEmptyString(proxyEnv.hostname),
+          port: Number(Util.validateEmptyString(proxyEnv.port)),
+          user: Util.validateEmptyString(proxyEnv.username),
```

Comment:
same thing, maybe `user` and `password` won't exist in proxy URL , if proxy is unauthenticated.

Reply:
Yes, in that case, user and password will be deleted. Please take a look at https://github.com/snowflakedb/snowflake-connector-nodejs/pull/938/files#diff-3fb720f70ea34fb2975fd6c38ac4b6b6131373be2abfa4760dfaae8a25daaee2R855.